bundles/org.openhab.binding.mideaac/src/main/java/org/openhab/binding/mideaac/internal/connection/ConnectionManager.java

```diff
@@ -0,0 +1,597 @@
+     */
+    public ConnectionManager(String ipAddress, int ipPort, int timeout, String key, String token, String cloud,
+            String email, String password, String deviceId, int version, boolean promptTone) {
+        this.deviceIsConnected = false;
+        this.ipAddress = ipAddress;
+        this.ipPort = ipPort;
+        this.timeout = timeout;
+        this.key = key;
+        this.token = token;
+        this.cloud = cloud;
+        this.deviceId = deviceId;
+        this.version = version;
+        this.promptTone = promptTone;
+        this.lastResponse = new Response(HexFormat.of().parseHex("C00042667F7F003C0000046066000000000000000000F9ECDB"),
+                version);
+        this.cloudProvider = CloudProvider.getCloudProvider(cloud);
+        this.security = new Security(cloudProvider);
+    }
+
+    private Socket socket = new Socket();
+    private InputStream inputStream = new ByteArrayInputStream(new byte[0]);
+    private DataOutputStream writer = new DataOutputStream(System.out);
+
+    /**
+     * Gets last response
+     * 
+     * @return byte array of last response
+     */
+    public Response getLastResponse() {
+        return this.lastResponse;
+    }
+
+    /**
+     * The socket is established with the writer and inputStream (for reading responses)
+     * V2 devices will proceed to send the poll or the set command.
+     * V3 devices will proceed to authenticate
+     * 
+     * @throws MideaConnectionException
+     * @throws MideaAuthenticationException
+     * @throws SocketTimeoutException
+     * @throws IOException
+     */
+    public synchronized void connect()
+            throws MideaConnectionException, MideaAuthenticationException, SocketTimeoutException, IOException {
+        logger.trace("Connecting to {}:{}", ipAddress, ipPort);
+
+        int maxTries = 3;
+        int retrySocket = 0;
+
+        // If resending command add delay to avoid connection rejection
+        // Suspect that the AC device needs a few seconds to clear.
+        if (!resend) {
+            try {
+                Thread.sleep(5000);
+            } catch (InterruptedException ex) {
+                logger.debug("An interupted error (resend command delay-connect) has occured {}", ex.getMessage());
+            }
+        }
+
+        // Open socket
+        // RetrySocket addresses the Timeout exception only, others exceptions end the thread. Same as HA python version
+        while (retrySocket < maxTries) {
+            try {
+                socket = new Socket();
+                socket.setSoTimeout(timeout * 1000);
+                socket.connect(new InetSocketAddress(ipAddress, ipPort), timeout * 1000);
+                break;
+            } catch (SocketTimeoutException e) {
+                retrySocket++;
+                if (retrySocket < maxTries) {
+                    try {
+                        Thread.sleep(5000);
+                    } catch (InterruptedException ex) {
```

Comment:
Not fully fixed this comment: https://github.com/openhab/openhab-addons/pull/17749/files#r1962228100
The interupt flag needs to be restored and exit quickly

Reply:
Made a change for review